GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -125,14 +126,14 @@ public virtual bool IsInDistress()
             return false; //return DistressCause;
         }
 
-        public virtual void TakeDamage(IAttackableUnit attacker, float damage, DamageType type, DamageSource source,
-            DamageText damageText)
+        public virtual void TakeDamage(IAttackableUnit attacker, Damage damage, DamageText damageText)
         {
             float defense = 0;
             float regain = 0;
+            float damageDealt = damage.DamageValue;
```

Comment:
This variable seems to be unused.

Reply:
Removed the line, it was a leftover from when I was using a class for damage (which are reference types) rather than a struct (which are value types). Thought I'd gotten rid of any occurrences of that happening, my bad.